flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -152,6 +153,12 @@
 
 	private TypeSerializer<T> eventSerializer;
 
+	/**
+	 * Flag indicating whether the matching status of the state machine has changed.
+	 */
+	@VisibleForTesting
```

Comment:
I think we need setter anyway (see AbstractKeyedCEPPatternOperator:277). So I would make it private.